### PR TITLE
[spark] Support StringEndsWith filter in SparkFilterConverter

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose

Linked issue: close https://github.com/apache/paimon/issues/3715

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/predicate/EndsWith.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            DataType type,
+            long rowCount,
+            Object min,
+            Object max,
+            Long nullCount,
+            Object patternLiteral) {
+        BinaryString minStr = (BinaryString) min;
+        BinaryString maxStr = (BinaryString) max;
+        BinaryString pattern = (BinaryString) patternLiteral;
+        return (minStr.endsWith(pattern) || minStr.compareTo(pattern) <= 0)
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/predicate/EndsWith.java:54]: Really works? Can you add more tests?

Endswith does not support range filtering very well.

Comment (xuzifu666) [reply paimon-common/src/main/java/org/apache/paimon/predicate/EndsWith.java:54]: OK，had add tests to test it. @JingsongLi

**Review by JingsongLi — COMMENTED** [review]
You should add tests in `PredicateTest`.

File: paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkFilterConverterTest.java [hunk 2]
```diff
@@ -192,6 +210,11 @@ public void testDate() {
 
     @Test
     public void testIgnoreFailure() {
+        testStartsWith();
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkFilterConverterTest.java:213]: No need modify this, this is only to test `convertIgnoreFailure`, regardless of the filter used

File: paimon-common/src/main/java/org/apache/paimon/predicate/EndsWith.java (hunk 1 above)
Comment (Zouxxyy) [reply paimon-common/src/main/java/org/apache/paimon/predicate/EndsWith.java]: +1 with @JingsongLi, does `endWith` really have any effect on range filtering?

**Comment by Zouxxyy** [comment]
By the way, since endwith has no range filtering capability, even we pass it to paimon, it has no filtering effect...

**Comment by Zouxxyy** [comment]
can add test for partition filering

File: paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkFilterConverterTest.java (hunk 2 above)
Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/test/java/org/apache/paimon/spark/SparkFilterConverterTest.java]: OK，removed the ut

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**